### ISSUE TITLE
Running sccache --start-server when the server is already running produces an error

### ISSUE THREAD

**Issue by staktrace** [mozilla/sccache#184]
If the server is already running in the background, and you try to start the server again manually using `sccache --start-server` it emits an error:

```
$ sccache --start-server
Starting sccache server...
error: Server startup error
caused by: Server startup failed: 1
```

This error message could be more useful, in that it could tell you that it's already running instead of looking like something went wrong.

This wasted some time during the investigation at https://github.com/rust-lang/rust/issues/42867#issuecomment-331606567

**Comment by luser** [comment]
Agreed. We should explicitly check for "socket in use" errors and do something smarter here.
